### ISSUE TITLE
Unable to open parameters database on fixed 3.6.4

### ISSUE THREAD

**Issue by noahboegli** [Infomaniak/desktop-kDrive#268]
Starting kDrive on linux (AppImage) gives the following error:

```
kDrive server starting
kDrive server initialization error: Unable to open parameters database.
```

It was apparently fixed for this version (as noted in #237), but at least for me the issue still arises.
Is there a manual step or database migration that must be done by an intermediary version ? 

(upgraded from 3.5.8, since for some reason the update notification does not work on linux, but that's a separate issue)

**Comment by ChristopheLarchier** [comment]
Hello, would it be possible to attach the log of the app? (located in /tmp/kDrive-logdir)
This might happen if you ran an older version of the app after migrating to the latest version.

**Comment by noahboegli** [comment]
Sure: [20240826_1054_kDrive.log](https://github.com/user-attachments/files/16746506/20240826_1054_kDrive.log)

I did not ran the old version per say, as the 3.6.4 version never managed to run. I basically shut down 3.5.8 and opened 3.6.4 and it crashed. Reverted back to 3.5.8  and it worked again.

**Comment by ngiger** [comment]
I had the same problem.

After running  `rm -rfv .config/kDrive/  /tmp/kDrive-logdir`

I have the following log [20240830_1052_kDrive.log](https://github.com/user-attachments/files/16815880/20240830_1052_kDrive.log)
Where it complains about a missing file  `/home/niklaus/squashfs-root/sync-exclude.lst
`
After `touch  /home/niklaus/squashfs-root/sync-exclude.lst` I  could  proceed to the login

**Comment by Feeback25** [comment]
Hello,
Same problem as noahboegli.
I am using Fedora 40. Kdrive has already worked on my machine.
For a few days, it does not launch.
Here is an extract from the logs:
"18:05:14 AppRun.wrapped: kDrive server initialization error: Unable to open parameters database.
18:05:14 AppRun.wrapped: kDrive server initialization error: Unable to open parameters database.
18:05:14 AppRun.wrapped: Warning: Ignoring WAYLAND_DISPLAY on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
18:05:14 AppRun.wrapped: kDrive server starting
18:05:13 systemd: Started app-gnome-kdrive-15001.scope - Application launched by gnome-shell."
Thanks for your help

**Comment by harryburli** [comment]
Hi,
same issue here using "kDrive-3.6.4.20240814-amd64.AppImage" on Linux Mint 21.3 Cinnamon.
do you know a work around?

![Bildschirmfoto vom 2024-09-12 15-52-26](https://github.com/user-attachments/assets/73efbf76-d93f-4c12-8caf-fc5f146d63dc)

I have the following log

[20240912_1605_kDrive.log](https://github.com/user-attachments/files/16980819/20240912_1605_kDrive.log)

Thank you in advance!

**Comment by herve-er** [comment]
Hello everyone,
After investigating, we have identified an issue with the migration of the parameters database from version 3.5.8 when skipping version 3.6.1. Unfortunately, we will not be able to provide a fix in the upcoming version (3.6.5) but it should be available in the next one.

However, we have found a workaround to resolve this issue on your installations:
1. Close kDrive if it is already open.
2. Launch the 3.5.8 AppImage.
3. Once it has started, close it.
4. Launch the 3.6.1 AppImage. (Download links below)
5. Once it has started, close it.
6. You should now be able to use kDrive 3.6.4.

kDrive 3.6.1 Linux AMD: https://kdrive.infomaniakgroup.com/app/share/100338/a41db454-e355-47c9-8e4c-c2789a23b164
kDrive 3.6.1 Linux ARM: https://kdrive.infomaniakgroup.com/app/share/100338/333a80e7-dfe6-4dd9-81ea-d3cdbf542baf

We apologize for the inconvenience.

**Comment by herve-er** [comment]
Hi @ngiger,

> After touch  /home/niklaus/squashfs-root/sync-exclude.lst I could proceed to the login

To avoid unexpected behavior, I suggest replacing the empty `sync-exclude.lst` file you created with the one bellow.
[sync-exclude-linux.zip](https://github.com/user-attachments/files/17028274/sync-exclude-linux.zip)

**Comment by harryburli** [comment]
Dear Infomaniak team,

after starting the 3.5.8 AppImage I get the following Popup:

Everytime I close it it pops up again immediately and blocks the
initialization of the login process. 

Same happens with version 3.6.1 AppImage

and version 3.6.4

Can you please check this issue.

Thanks in advance.

Harald

_FREUNDLICHE GRÜSSE__,_

_HARALD WEBER_

Am 2024-09-17T14:33:26.000+02:00 hat Herve Eruam
***@***.***> geschrieben:

**Comment by herve-er** [comment]
Hello @harryburli,
It seems that the image of the popup was not attached to your comment. Could you please provide us with it?

Thank you.

**Comment by harryburli** [comment]
Hi Herve,

yes of course additionally I send you the log files to.
Please give me feedback if you have received the files.

Thank you for your support.

_BEST REGARDS,_

_HARALD WEBER_

Am 2024-09-18T08:36:32.000+02:00 hat Herve Eruam
***@***.***> geschrieben:

**Comment by herve-er** [comment]
Unfortunately, the attached files from your emails were not transferred to the GitHub discussion. Could you please upload them directly from the GitHub discussion ( https://github.com/Infomaniak/desktop-kDrive/issues/268 )?

Thank you.

**Comment by harryburli** [comment]
Jep!
[20240917_2057_kDrive.log.gz](https://github.com/user-attachments/files/17043552/20240917_2057_kDrive.log.gz)
[20240917_2059_kDrive.log](https://github.com/user-attachments/files/17043553/20240917_2059_kDrive.log)
![Bildschirmfoto vom 2024-09-17 20-45-00](https://github.com/user-attachments/assets/3c0979b2-ee58-41a0-9b23-a941beefa4d2)

**Comment by herve-er** [comment]
Hi @harryburli,
Thank you for the additional information.

Since this is unrelated to the parameters database issue, I have opened a new one for you --> https://github.com/Infomaniak/desktop-kDrive/issues/307

**Comment by noahboegli** [comment]
> Hello everyone, After investigating, we have identified an issue with the migration of the parameters database from version 3.5.8 when skipping version 3.6.1. Unfortunately, we will not be able to provide a fix in the upcoming version (3.6.5) but it should be available in the next one.
> 
> However, we have found a workaround to resolve this issue on your installations:
> 
>     1. Close kDrive if it is already open.
> 
>     2. Launch the 3.5.8 AppImage.
> 
>     3. Once it has started, close it.
> 
>     4. Launch the 3.6.1 AppImage. (Download links below)
> 
>     5. Once it has started, close it.
> 
>     6. You should now be able to use kDrive 3.6.4.
> 
> 
> kDrive 3.6.1 Linux AMD: https://kdrive.infomaniakgroup.com/app/share/100338/a41db454-e355-47c9-8e4c-c2789a23b164 kDrive 3.6.1 Linux ARM: https://kdrive.infomaniakgroup.com/app/share/100338/333a80e7-dfe6-4dd9-81ea-d3cdbf542baf
> 
> We apologize for the inconvenience.

@herve-er  Thank you ! I can confirm the upgrad path works. 
I don't know if you want to keep this issue open for reference, I'll let you guys be the judges of that.

**Comment by herve-er** [comment]
@noahboegli Thank you for the feedback!
We will keep this issue open until we develop a proper fix.

**Comment by jlebleu** [comment]
> > Hello everyone, After investigating, we have identified an issue with the migration of the parameters database from version 3.5.8 when skipping version 3.6.1. Unfortunately, we will not be able to provide a fix in the upcoming version (3.6.5) but it should be available in the next one.
> > However, we have found a workaround to resolve this issue on your installations:
> > ```
> > 1. Close kDrive if it is already open.
> > 
> > 2. Launch the 3.5.8 AppImage.
> > 
> > 3. Once it has started, close it.
> > 
> > 4. Launch the 3.6.1 AppImage. (Download links below)
> > 
> > 5. Once it has started, close it.
> > 
> > 6. You should now be able to use kDrive 3.6.4.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > kDrive 3.6.1 Linux AMD: https://kdrive.infomaniakgroup.com/app/share/100338/a41db454-e355-47c9-8e4c-c2789a23b164 kDrive 3.6.1 Linux ARM: https://kdrive.infomaniakgroup.com/app/share/100338/333a80e7-dfe6-4dd9-81ea-d3cdbf542baf
> > We apologize for the inconvenience.
> 
> @herve-er Thank you ! I can confirm the upgrad path works. I don't know if you want to keep this issue open for reference, I'll let you guys be the judges of that.

 Did the same, works for me also, kdrive is restarted!!! Yess

**Comment by claudep** [comment]
Just to chime in here, I had to debug a similar situation today with someone coming from a 3.3.x version of kmeet, and the same issue with the config db not upgradable (some SQLite column missing). We resolved the situation by removing the old config database, but it would be nice if you could keep an history of the config db migrations and detect which migration(s) have to be applied.
